library/kubevirt_vm.py

```diff
@@ -11,26 +15,102 @@
 VERSION = 'v1alpha1'
 
 
+ANSIBLE_METADATA = {'metadata_version': '1.1',
+                    'status': ['preview'],
+                    'supported_by': 'community'}
+
 DOCUMENTATION = '''
+---
 module: kubevirt_vm
-short_description: Handles kubevirt vms
+short_description: Manage KubeVirt VMs
 description:
-    - Longer description of the module
-    - Use name, namespace and src
-version_added: "0.1"
-author: "Karim Boumedhel, @karmab"
+    - Create or delete a KubeVirt VM
+version_added: "2.4.x"
+author:
+    - Karim Boumedhel (@karmab)
+    - Irina Gulina (@alexxa)
+options:
+    state:
+        description:
+            - Whether to create (C(present)) or delete (C(absent)) a VM.
+        required: false
+        default: "present"
+        choices: ["present", "absent"]
+    name:
+        description:
+            - Name of a VM.
+        required: true
+    namespace:
+        description:
+            - Namespace to add a VM to or delete from.
+        required: true
+    wait:
+        description:
+            - Wait for a VM to be running.
+        type: bool
+        required: false
+        default: 'no'
+    timeout:
+        description:
+            - Maximum number of seconds to wait for.
+        type: int
+        required: false
+        default: "20"
+    memory:
+        description
+            -
+        required: false
+        default: "512M"
+    disk:
+        description:
+            - Attach a volume as a disk to a VM.
+        required: false
+    pvc:
+        description:
+            - Name of a PersistentVolumeClaim existing in the same namespace to use as a base disk for a VM.
+        required: false
+    lun:
+        description:
+            - Attach a volume as a LUN to a VM.
+        type: int
+        required: false
+        default: "3"
+    iqn:
+        description:
+            -
+        required: false
+        default: "iqn.2017-01.io.kubevirt:sn.42"
+    target:
```

Comment:
lun, iqn, target -- all should be removed, as we only support pvc since 0.4.0 (direct iscsi support was dropped)